concrete/jobs/index_search.php

```diff
@@ -47,4 +102,62 @@ protected function pagesToQueue()
         }
     }
 
+    /**
+     * Get a list of sites to remove from the search index
+     *
+     * @return int[]
+     */
+    protected function sitesToRemove()
+    {
+        return [];
+    }
+
+    /**
+     * Get a list of files to remove from the search index
+     * @return int[]
+     */
+    protected function filesToRemove()
+    {
+        return [];
+    }
+
+    /**
+     * Get a list of users to remove from the search index
+     *
+     * @return int[]
+     */
+    protected function usersToRemove()
+    {
+        return [];
+    }
+
+    /**
+     * Get a list of pages to be removed from the search index
+     *
+     * @return int[];
+     */
+    protected function pagesToRemove()
+    {
+        $qb = $this->connection->createQueryBuilder();
+
+        // Find all pages that need to be removed from the index
+        $query = $qb
+            ->select('p.cID')
+            ->from('Pages', 'p')
+            ->leftJoin('p', 'CollectionSearchIndexAttributes', 'a', 'p.cID = a.cID')
+            ->leftJoin('p', 'PageSearchIndex', 's', 'p.cID = s.cID')
+            ->where('s.cDateLastIndexed IS NOT NULL')
+            ->andWhere($qb->expr()->orX(
+                'p.cIsActive != 1',
+                $qb->expr()->andX(
+                    'a.ak_exclude_search_index IS NOT NULL',
```

Comment:
I agree with @Mnkras: we can't assume that users don't change the default indexes...